src/common/tags.h

```diff
@@ -78,11 +78,14 @@ void dt_tag_rename(const guint tagid, const gchar *new_tagname);
 gboolean dt_tag_exists(const char *name, guint *tagid);
 
 /** attach a tag on selected images. tagid id of tag to attach. imgid the image
- * id to attach tag to, if < 0 selected images are used. */
-void dt_tag_attach(const guint tagid, const gint imgid, const gboolean undo_on, const gboolean group_on);
+ * id to attach tag to, if < 0 selected images are used. Returns FALSE if tag was already present */
```

Comment:
Same as before. The requirement for the caller is to have an existing tag before attaching it.